src/Module/Filer/RemoveTag.php

```diff
@@ -53,28 +91,27 @@ protected function content(array $request = []): string
 			$type = Post\Category::FILE;
 		}
 
-		$logger->info('Filer - Remove Tag', [
+		$this->logger->info('Filer - Remove Tag', [
 			'term' => $term,
 			'item' => $item_id,
 			'type' => $type
 		]);
 
-		if ($item_id && strlen($term)) {
-			$item = Post::selectFirst(['uri-id'], ['id' => $item_id]);
-			if (!DBA::isResult($item)) {
-				return '';
-			}
-			if (!Post\Category::deleteFileByURIId($item['uri-id'], local_user(), $type, $term)) {
-				notice(DI::l10n()->t('Item was not removed'));
-			}
-		} else {
-			notice(DI::l10n()->t('Item was not deleted'));
+		if (!$item_id || !strlen($term)) {
+			$this->systemMessages->addNotice($this->l10n->t('Item was not deleted'));
+			return 401;
 		}
 
-		if ($type == Post\Category::FILE) {
-			DI::baseUrl()->redirect('filed?file=' . rawurlencode($term));
+		$item = Post::selectFirst(['uri-id'], ['id' => $item_id]);
+		if (!DBA::isResult($item)) {
+			return 404;
 		}
 
-		return '';
+		if (!Post\Category::deleteFileByURIId($item['uri-id'], local_user(), $type, $term)) {
+			$this->systemMessages->addNotice($this->l10n->t('Item was not removed'));
+			return 500;
+		}
+
+		return 200;
```

Comment:
Are there equivalent constants for HTTP codes we could use?

Reply:
Not that I know of 🤔